consensus/types/src/light_client_update.rs

```diff
@@ -259,6 +248,150 @@ impl<E: EthSpec> LightClientUpdate<E> {
+
+        let new_active_participants = new.sync_aggregate().sync_committee_bits.num_set_bits();
+        let prev_active_participants = self.sync_aggregate().sync_committee_bits.num_set_bits();
+
+        let new_has_super_majority =
+            new_active_participants.safe_mul(3)? >= max_active_participants.safe_mul(2)?;
+        let prev_has_super_majority =
+            prev_active_participants.safe_mul(3)? >= max_active_participants.safe_mul(2)?;
+
+        if new_has_super_majority != prev_has_super_majority {
+            return Ok(new_has_super_majority & !prev_has_super_majority);
+        }
+
+        if !new_has_super_majority && new_active_participants != prev_active_participants {
+            return Ok(new_active_participants > prev_active_participants);
+        }
+
+        let new_attested_header_slot = match new {
+            LightClientUpdate::Altair(update) => update.attested_header.beacon.slot,
+            LightClientUpdate::Capella(update) => update.attested_header.beacon.slot,
+            LightClientUpdate::Deneb(update) => update.attested_header.beacon.slot,
+        };
+
+        let new_finalized_header_slot = match new {
+            LightClientUpdate::Altair(update) => update.finalized_header.beacon.slot,
+            LightClientUpdate::Capella(update) => update.finalized_header.beacon.slot,
+            LightClientUpdate::Deneb(update) => update.finalized_header.beacon.slot,
+        };
+
+        let prev_attested_header_slot = match self {
+            LightClientUpdate::Altair(prev) => prev.attested_header.beacon.slot,
+            LightClientUpdate::Capella(prev) => prev.attested_header.beacon.slot,
+            LightClientUpdate::Deneb(prev) => prev.attested_header.beacon.slot,
+        };
+
+        let prev_finalized_header_slot = match self {
+            LightClientUpdate::Altair(prev) => prev.finalized_header.beacon.slot,
+            LightClientUpdate::Capella(prev) => prev.finalized_header.beacon.slot,
+            LightClientUpdate::Deneb(prev) => prev.finalized_header.beacon.slot,
+        };
+
+        let new_attested_header_sync_committee_period =
+            compute_sync_committee_period_at_slot::<E>(new_attested_header_slot, chain_spec)?;
+
+        let new_signature_slot_sync_committee_period =
+            compute_sync_committee_period_at_slot::<E>(*new.signature_slot(), chain_spec)?;
+
+        let prev_attested_header_sync_committee_period =
+            compute_sync_committee_period_at_slot::<E>(prev_attested_header_slot, chain_spec)?;
+
+        let prev_signature_slot_sync_committee_period =
+            compute_sync_committee_period_at_slot::<E>(*self.signature_slot(), chain_spec)?;
+
+        // Compare presence of relevant sync committee
+        let new_has_relevant_sync_committee = !new.is_next_sync_committee_branch_empty()
+            && (new_attested_header_sync_committee_period
+                == new_signature_slot_sync_committee_period);
```

Comment:
Can also condense this logic into a method `attested_header_in_signature_period`